lib/util/jsonUtils.ts

```diff
@@ -166,7 +166,11 @@ export const getFilePositionFromJsonPath = (
   const pathArr = jsonPathToArray(jsonPath.substr(1));
   const newPathArr = pathArr.slice(0);
   const index = newPathArr.findIndex((str) => str.includes("/providers/Microsoft"));
-  if (index !== -1 && newPathArr[index + 1] !== undefined) {
```

Comment:
can you add comment about the error path to be processed?

Reply:
can you update comment as: when jsonPath='/providers/Microsoft.Provider/resource', the split pathArr will be ['/providers/Microsoft','Provider/resource'].  Only in this case, these two elements in the array need to be composed together by '.'. So restrict the condition to the path element ends with /providers/Microsoft.